vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java

```diff
@@ -347,55 +349,82 @@ public void addHeadersToProxyRequest(StringComparisonRule comparisonRule, String
-            httpHeaders.add(headers);
-            return mockedRequest;
-        });
+        mockHttpRequests(Set.of(), comparisonRule, url, responseCode, content, headers);
     }
 
     /**
      * The step allows to mock HTTP request, and provide within response status code and headers.
      * Once the request is matched by URL comparison rule response will be returned. In this case no actual request will
      * be executed.
      * @param comparisonRule String comparison rule: "is equal to", "contains", "does not contain", "matches"
-     * @param url The string value of URL to match comparison rule
-     * @param responseCode The code will be used in response
-     * @param headers ExamplesTable representing the list of the headers with columns "name" and "value" specifying
-     * HTTP header names and values respectively
+     * @param url            The string value of URL to match comparison rule
+     * @param responseCode   The code will be used in response
+     * @param headers        ExamplesTable representing the list of the headers with columns "name" and
+     *                       "value" specifying HTTP header names and values respectively
      */
     @When("I mock HTTP responses with request URL which $comparisonRule `$url` using response code `$responseCode`"
             + " and headers:$headers")
     public void mockHttpRequests(StringComparisonRule comparisonRule, String url, int responseCode,
             DefaultHttpHeaders headers)
     {
-        applyUrlFilter(comparisonRule, url, request -> {
+        applyUrlFilter(List.of(m -> comparisonRule.createMatcher(url).matches(m.getUrl())), request -> {
             DefaultHttpResponse mockedRequest =
                 new DefaultHttpResponse(request.protocolVersion(), HttpResponseStatus.valueOf(responseCode));
             mockedRequest.headers().add(headers);
             return mockedRequest;
         });
     }
 
+    /**
+     * The step allows to mock HTTP request of specified method, and provide within response status code,
+     * headers and content. Once the request is matched by method and URL comparison rule response will be returned.
+     * In this case no actual request will be executed.
+     * @param httpMethods    The "or"-separated set of HTTP methods to filter by, e.g. 'GET or POST or PUT'
+     * @param comparisonRule String comparison rule: "is equal to", "contains", "does not contain", "matches"
+     * @param url            The string value of URL to match comparison rule
+     * @param responseCode   The code will be used in response
+     * @param content        The content will be used in the response
+     * @param headers        ExamplesTable representing the list of the headers with columns "name" and
+     *                       "value" specifying HTTP header names and values respectively
+     */
+    @When(value = "I mock HTTP responses with HTTP $httpMethods and request URL which $comparisonRule `$url` using"
```

Comment:
When I mock HTTP GET or POST responses....?

Reply:
> When I mock HTTP GET or POST responses....?

`When I mock HTTP $httpMethods responses with request URL which $comparisonRule `$url` using response code `$responseCode`, content `$payload` and headers:$headers`
?